ios/brave-ios/Sources/Brave/Frontend/Settings/General/SearchEngines/Custom Engines/CustomEngineViewController.swift

```diff
@@ -71,6 +73,19 @@ class CustomEngineViewController: UIViewController {
 
   private var engineToBeEdited: OpenSearchEngine?
   var customEngineActionType: CustomEngineActionType = .add
+  var doneButtonStatus: DoneButtonStatus = .disabled {
+    didSet {
+      if doneButtonStatus == .disabled {
+        isAutoAddEnabled = false
+      }
+      ensureMainThread {
```

Comment:
This variable is already MainActor isolated due to it being part of a UIViewController subclass, so this hop shouldn't be needed

Reply:
i'm getting an exception of modification on layout not on main without this `ensureMainThread`